packages/x-data-grid-pro/src/hooks/features/lazyLoader/useGridLazyLoader.ts

```diff
@@ -104,11 +82,7 @@ export const useGridLazyLoader = (
     firstRowToRender: 0,
     lastRowToRender: 0,
   });
-  const { lazyLoading } = (props.experimentalFeatures ?? {}) as GridExperimentalProFeatures;
-  const isDisabled = isLazyLoadingDisabled({
-    lazyLoadingFeatureFlag: lazyLoading,
-    rowsLoadingMode: props.rowsLoadingMode,
-  });
+  const isDisabled = props.rowsLoadingMode !== 'server';
```

Comment:
Would `rowLoadingMode = 'server'` be enough as a feature flag? Technically it should be the case for other server-side row-loading use cases too (like infinite loading for example), no?
Wdyt about adding a feature flag like (`props.lazyLoading`) to enable the lazy loading feature explicitly?

Reply:
> Technically it should be the case for other server-side row-loading use cases too (like infinite loading for example), no?

Infinite loading requires `rowsLoadingMode="client"`:
https://github.com/mui/mui-x/blob/ed9073da1e2849bf2f7b9358e0a1156bbb62bfc0/packages/x-data-grid-pro/src/hooks/features/infiniteLoader/useGridInfiniteLoader.tsx#L41
So `rowsLoadingMode="server"` is exclusive to lazy loading.
I don't see much value in changing this right now.